### PR TITLE
Add List-Unsubscribe email header

### PR DESCRIPTION
**Pull request by c960657**
Unsubscribe links were added to the email footer in #25378.

To allow for easier unsubscription, add support for `List-Unsubscribe` as described in [RFC 2369](https://datatracker.ietf.org/doc/html/rfc2369) and the one-click extension defined in [RFC 8058](https://datatracker.ietf.org/doc/html/rfc8058). This will show an _Unsubscribe_ button in the native UI, making it easier for user to find.

`List-Unsubscribe` with `https://` links is supported by [at least one major email client/provider](https://www.litmus.com/blog/the-ultimate-guide-to-list-unsubscribe) (Gmail). Outlook.com and iOS Mail only support `mailto:` in `List-Unsubscribe` which is outside the scope of this PR.

This is difficult to test in practice, because Gmail only shows the unsubscribe button for [senders with high reputation](https://stackoverflow.com/questions/28497332/gmail-unsubscribe-link-does-not-appear). So I have no been able to verify that it actuall works with Gmail.

The headers added are:
- `List-Unsubscribe`: Link to the unsubscribe confirmation page.
- `List-Unsubscribe-Post`: Extension that allows bypassing the manual confirmation page (one-click).
- `List-ID`: Allows filtering emails. [Supported in Gmail](http://googlesystem.blogspot.com/2007/11/filter-messages-from-mailing-list-in.html).
- `List-Archive`: Link to the notifications list in the React frontend.
- `List-Subscribe`: Link to the notification settings page.

The latter two are not supported by any mail clients that I know of, but they don't add any code complexity.

### CODE REVIEW

File: app/controllers/mail_subscriptions_controller.rb [hunk 1]
```diff
@@ -9,6 +9,8 @@ class MailSubscriptionsController < ApplicationController
   before_action :set_user
   before_action :set_type
 
+  protect_from_forgery
```
Comment (ClearlyClaire) [inline app/controllers/mail_subscriptions_controller.rb:12]: What is this change for? `ApplicationController` already sets `protect_from_forgery with: :exception`

File: app/mailers/notification_mailer.rb [hunk 2]
```diff
@@ -71,6 +73,14 @@ def set_account
     @account = @notification.from_account
   end
 
+  def list_headers
+    headers['List-ID'] = "<#{@type}.#{Rails.configuration.x.local_domain}>"
+    headers['List-Archive'] = "<#{web_url('notifications')}>"
+    headers['List-Subscribe'] = "<#{settings_preferences_notifications_url}>"
+    headers['List-Unsubscribe'] = "<#{@unsubscribe_url}>"
+    headers['List-Unsubscribe-Post'] = 'List-Unsubscribe=One-Click'
```
Comment (ClearlyClaire) [inline app/mailers/notification_mailer.rb:81]: I'm not sure we want *all* of these headers. Also the `List-ID` could possibly use the username, as someone may have multiple account on the same server.

File: app/controllers/mail_subscriptions_controller.rb (hunk 1 above)
Comment (c960657) [reply app/controllers/mail_subscriptions_controller.rb]: The POST request with `List-Unsubscribe=One-Click` made by the email client should not throw an exception. 

`protect_from_forgery` without `with` defaults to `with: : null_session`. I have now changed this to `skip_forgery_protection` to make it more explicit that we don't care about CSRF here.

File: app/mailers/notification_mailer.rb (hunk 2 above)
Comment (c960657) [reply app/mailers/notification_mailer.rb]: Ok, I have removed `List-Archive` and `List-Subscribe`. We can easily add them again if popular email clients adds support for them.

Good point about `List-ID`. It now also includes the username.

File: app/controllers/mail_subscriptions_controller.rb (hunk 1 above)
Comment (ClearlyClaire) [reply app/controllers/mail_subscriptions_controller.rb]: Ah, I did not know the email client itself would do a POST request. That makes sense, although no session should be needed, so `:null_session` should be fine.

Comment (c960657) [reply app/controllers/mail_subscriptions_controller.rb]: Ok, reverted to `protect_from_forgery with: :null_session` :-)

File: app/controllers/mail_subscriptions_controller.rb [hunk 3]
```diff
@@ -9,6 +9,8 @@
   before_action :set_user
   before_action :set_type
 
+  protect_from_forgery with: :null_session
```
[the remark this answers is not in the document]
Comment (c960657) [reply app/controllers/mail_subscriptions_controller.rb:12]: This change is intentional. Do I need to do something to mute this error?

**Review by ClearlyClaire — APPROVED** [review]

**Pull request merged**